sw/cheri/common/boot.S

```diff
@@ -3,6 +3,109 @@
 # SPDX-License-Identifier: Apache-2.0
 .include "assembly-helpers.s"
 
+  .section .vectors, "ax"
+	.p2align 4
+  .global _trap_vector
+  .type _trap_vector, @function
+_trap_vector:
+// Save all the register to the stack
+  cincoffset csp, csp, -(8 * 28)
+  csc   cra,  1 * 8(csp)
+  csc   ct0,  2 * 8(csp)
+  csc   ct1,  3 * 8(csp)
+  csc   ct2,  4 * 8(csp)
+  csc   cs0,  5 * 8(csp)
+  csc   cs1,  6 * 8(csp)
+  csc   ca0,  7 * 8(csp)
+  csc   ca1,  8 * 8(csp)
+  csc   ca2,  9 * 8(csp)
+  csc   ca3, 10 * 8(csp)
+  csc   ca4, 11 * 8(csp)
+  csc   ca5, 12 * 8(csp)
+  csc   ca6, 13 * 8(csp)
+  csc   ca7, 14 * 8(csp)
+  csc   cs2, 15 * 8(csp)
+  csc   cs3, 16 * 8(csp)
+  csc   cs4, 17 * 8(csp)
+  csc   cs5, 18 * 8(csp)
+  csc   cs6, 19 * 8(csp)
+  csc   cs7, 20 * 8(csp)
+  csc   cs8, 21 * 8(csp)
+  csc   cs9, 22 * 8(csp)
+  csc  cs10, 23 * 8(csp)
+  csc  cs11, 24 * 8(csp)
+  csc   ct3, 25 * 8(csp)
+  csc   ct4, 26 * 8(csp)
+  csc   ct5, 27 * 8(csp)
+  csc   ct6, 28 * 8(csp)
+
+.L_software:
+  csrr t0, mip
```

Comment:
I think this whole logic should be moved to some C code.

Reply:
done.